pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -194,7 +194,7 @@ stdenv.mkDerivation ({
     runHook preSetupCompilerEnvironment
 
     echo "Build with ${ghc}."
-    export PATH="${ghc}/bin:$PATH"
+    export PATH="${if ghc.isGhcjs or false then ghcEnv else ghc}/bin:$PATH"
```

Comment:
The [original commit](https://github.com/NixOS/nixpkgs-channels/commit/09c3d04b0e53d65f60569fc01698decff3a657a3) reads "This helps support compiler plugins." @ryantrinkle Could you give a short description of why this is so (and why we would want `ghcEnv` for ghcjs but not native ghc)? I would also suggest putting the rationale as a comment somewhere near this line.